lib/config.js

```diff
@@ -50,23 +50,29 @@ const paramsSchema = Joi.object({
   auth0Logout: Joi.boolean().optional().default(false),
   authorizationParams: Joi.object({
     response_type: Joi.string().optional().valid('id_token', 'code id_token', 'code').default('id_token'),
-    scope: Joi.string().optional().default('openid profile email'),
-    response_mode: Joi.alternatives([
-      Joi.string().optional(),
-      Joi.allow(null).optional()
-    ]).default((parent) => parent.response_type.includes('token') ? 'form_post' : undefined),
+    scope: Joi.string().optional().pattern(/\bopenid\b/, 'contains openid').default('openid profile email'),
+    response_mode: Joi.string().optional().when('response_type', {
+      is: 'code',
+      then: Joi.valid('query', 'form_post').default('query'),
```

Comment:
when code, leave `undefined` (i.e. no default) to omit the parameter from being sent, all servers are required to respond with a query automatically. For all other response types we send the value form_post explicitly because it's not the default to be used.

Reply:
1da5afe